PluginsAndFeatures/azure-toolkit-for-intellij/config/checkstyle/checkstyle.xml

```diff
@@ -1,114 +1,247 @@
+        <property name="lineSeparator" value="lf"/>
+    </module>
+
+    <module name="RegexpSingleline">
+        <!-- Checks that FIXME is not used in comments.  TODO is preferred.
+        -->
+        <property name="format" value="((//.*)|(\*.*))FIXME"/>
+        <property name="message"
+                  value='TODO is preferred to FIXME.  e.g. "TODO(johndoe): Refactor when v2 is released."'/>
+    </module>
+
+
     <!-- No trailing spaces allowed -->
     <module name="RegexpSingleline">
         <property name="format" value="\s+$"/>
         <property name="message" value="Line has trailing spaces."/>
     </module>
 
-    <module name="TreeWalker">
-        <!-- Default CheckStyle Whitespace rules seem reasonable to use. -->
-        <module name="EmptyForIteratorPad"/>
-        <module name="MethodParamPad"/>
-        <module name="NoWhitespaceAfter"/>
-        <module name="NoWhitespaceBefore"/>
-        <module name="OperatorWrap"/>
-        <module name="ParenPad"/>
-        <module name="TypecastParenPad"/>
-        <module name="WhitespaceAfter"/>
-        <module name="WhitespaceAround"/>
+    <!-- CRLF is not allowed -->
+    <module name="RegexpMultiline">
+        <property name="format" value="(?s:(\r\n|\r).*)"/>
+        <property name="message" value="CRLF and CR line endings are prohibited, but this file uses them."/>
+    </module>
 
-        <!-- Modifier Order as specified by Java Language Specifications. -->
-        <module name="ModifierOrder"/>
+    <module name="RegexpSingleline">
+        <!-- Checks that TODOs are named.  (Actually, just that they are followed
+             by an open paren.)
+        -->
+        <property name="format" value="((//.*)|(\*.*))TODO [^(]"/>
+        <property name="message"
+                  value='All TODOs should be named.  e.g. "TODO (johndoe): Refactor when v2 is released."'/>
+    </module>
 
-        <!-- Empty blocks allowed, as long as they have explicit comments stating why they are empty.  -->
-        <module name="EmptyBlock">
-            <property name="option" value="text"/>
-        </module>
+    <module name="LineLengthCheck">
+        <property name="fileExtensions" value="java" />
+        <!-- Checks if a line is too long, default value of 120 is too short -->
+        <property name="max" value="${com.puppycrawl.tools.checkstyle.checks.sizes.LineLength.max}" default="120"/>
+        <property name="severity" value="error"/>
 
+        <!--
+            The default ignore pattern exempts the following elements:
+            - import statements
+            - long URLs inside comments
+        -->
 
-        <module name="LeftCurly"/>
-        <module name="NeedBraces"/>
-        <module name="RightCurly"/>
+        <property name="ignorePattern"
+                    value="${com.puppycrawl.tools.checkstyle.checks.sizes.LineLength.ignorePattern}"
+                    default="^(package .*;\s*)|(import .*;\s*)|( *\* *https?://.*)$"/>
```

Comment:
Can we add a regex, such as `\w+://.*` to ignore URI string (like: wasbs://xxx, adfs://xxx, webhdfs://xxx)?